src/components/buttons/_private/ButtonBase/ButtonBase.tsx

```diff
@@ -140,20 +143,24 @@ const ButtonBase = (props: IButtonBaseProps) => {
 	const LeftIcon = leftIcon;
 	const RightIcon = rightIcon;
 
-	const [ isActive, setIsActive ] = React.useState(false);
+	const [ isActive, setIsActive ] = React.useState(active);
 
 	const handleKeyDown = (e: React.KeyboardEvent) => {
-		if (e.key === 'Enter') {
+		if (e.key === 'Enter' && !active) {
 			setIsActive(true);
 		}
 	}
 
 	const handleKeyUp = (e: React.KeyboardEvent) => {
-		if (e.key === 'Enter') {
+		if (e.key === 'Enter' && !active) {
```

Comment:
Should this check be `if (e.key === 'Enter' && active) { ... }` since you follow with setting `active` to false afterwards?

Reply:
I follow with setting `isActive` to false. `active` is a prop for forcing the button to be active, like `disabled`. This code is basically saying only let the enter key add or remove the `_active` class if the parent isn't specifying that the button should be active.

Reply:
I'm following now. I got `active` and `isActive` mixed up I think.